server/src/main/java/org/opensearch/common/cache/store/listeners/dispatchers/StoreAwareCacheListenerDispatcherDefaultImpl.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.store.listeners.dispatchers;
+
+import org.opensearch.common.Nullable;
+import org.opensearch.common.cache.store.StoreAwareCacheRemovalNotification;
+import org.opensearch.common.cache.store.enums.CacheStoreType;
+import org.opensearch.common.cache.store.listeners.EventType;
+import org.opensearch.common.cache.store.listeners.StoreAwareCacheEventListenerConfiguration;
+
+/**
+ * Default implementation of event listener dispatcher.
+ * @param <K> Type of key.
+ * @param <V> Type of value.
+ */
+public class StoreAwareCacheListenerDispatcherDefaultImpl<K, V> implements StoreAwareCacheEventListenerDispatcher<K, V> {
```

Comment:
Similarly don't see a need of this `Dispatcher` based model, may be I am missing something here but why can't we just use the `StoreAwareCacheEventListener` directly wherever this dispatcher is used ?

Reply:
I created this dispatcher model with two reasons:
1. Having a single place where we can add the logic whether we want to send events for certain evenTypes. Otherwise every listener implementation will have to make those redundant changes.
2. I am looking to add a capability to offload listener based events to a separate thread as we don't need to add that overhead in the main path. This class can provide that as well.

Let me know what you think.

Reply:
Hmm, I think we will remove these dispatcher changes for now. Looks redundant to me when I think of it again.